### PR TITLE
Fix RebaseCustom for TK2 gates

### PR DESCRIPTION
**Pull request by sjdilkes**

### CODE REVIEW

File: tket/src/Transformations/Rebase.cpp [hunk 1]
```diff
@@ -122,6 +122,7 @@ static bool standard_rebase_via_tk2(
         tk2_replacement) {
   bool success = false;
   VertexSet bin;
+  std::cout << "It's definitely in the TK2 version. " << std::endl;
```
Comment (cqc-alec) [inline tket/src/Transformations/Rebase.cpp:125]: Delete.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: tket/src/Transformations/Rebase.cpp (hunk 1 above)
Comment (sjdilkes) [reply tket/src/Transformations/Rebase.cpp:125]: haha oops.

File: tket/src/Transformations/Rebase.cpp [hunk 2]
```diff
@@ -133,11 +134,11 @@ static bool standard_rebase_via_tk2(
       op = cond.get_op();
     }
     OpType type = op->get_type();
-    if (allowed_gates.contains(type) || type == OpType::TK2 ||
-        type == OpType::Barrier)
-      continue;
+    if (allowed_gates.contains(type) || type == OpType::Barrier) continue;
     // need to convert
+    std::cout << "Converting to TK2. " << std::endl;
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply tket/src/Transformations/Rebase.cpp:139]: done

File: tket/src/Transformations/Rebase.cpp (hunk 1 above)
Comment (sjdilkes) [reply tket/src/Transformations/Rebase.cpp:125]: done

File: pytket/tests/predicates_test.py [hunk 3]
```diff
@@ -902,6 +902,24 @@ def test_PeepholeOptimise2Q() -> None:
...
+
+    to_phase_gates = RebaseCustom(
+        {OpType.Rx, OpType.Rz, OpType.XXPhase, OpType.YYPhase, OpType.ZZPhase},
+        tk2_replacement=_tk2_to_phase,
+        tk1_replacement=_tk1_to_phase,
+    )
+
+    tk2_c = Circuit(2).TK2(0.123, 0.5634, 0.2345, 0, 1)
+    assert to_phase_gates.apply(tk2_c)
+    assert len(tk2_c.get_commands()) == 11
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply pytket/tests/predicates_test.py:920]: done

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**